src/Lean/Util/Trace.lean

```diff
@@ -250,6 +251,13 @@ instance [always : MonadAlwaysExcept ε m] [STWorld ω m] [BEq α] [Hashable α]
     MonadAlwaysExcept ε (MonadCacheT α β m) where
   except := let _ := always.except; inferInstance
 
+/-- Run the provided action `k`, and log its execution within a trace node.
+
+The message is produced after the action completes, and has access to its return value.
+If profiling is enabled, this will also log the runtime of `k`.
+
+The `cls`, `collapsed`, and `tag` arguments are fowarded to the constructor of `TraceData`.
+-/
```

Comment:
Maybe also contrast whith `withTraceNode'`.

Maybe show the idiom with `(msg := (return m!"{exceptEmoji \.} Blah blah")` for a concise parameter.

Reply:
Done.